### PR TITLE
Code Refactoring for CommonName

### PR DESCRIPTION
**Pull request by kaituo**
### Description
In this pull request, I have refactored the code related to shared names in both AD and forecasting modules to CommonNames. Additionally, the previously used CommonName has been renamed to ADCommonName. For the Forecasting module, I have introduced new names in ForecastCommonNames.

Testing done:
* gradle build

### Issues Resolved
https://github.com/opensearch-project/anomaly-detection/issues/866

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/ad/constant/ADCommonName.java [hunk 1]
```diff
@@ -11,17 +11,16 @@
...
-public class CommonName {
+public class ADCommonName {
     // ======================================
     // Index name
     // ======================================
     // index name for anomaly checkpoint of each model. One model one document.
     public static final String CHECKPOINT_INDEX_NAME = ".opendistro-anomaly-checkpoints";
     // index name for anomaly detection state. Will store AD task in this index as well.
     public static final String DETECTION_STATE_INDEX = ".opendistro-anomaly-detection-state";
-    // TODO: move other index name here
```
Comment (sudiptoguha) [inline src/main/java/org/opensearch/ad/constant/ADCommonName.java:24]: Any plans to change the "opendistro-anomaly" moniker? Isn't "OpenSearch" more appropriate? Can be a different PR though. OpenDistro may indicate that this is serving dated models (which is not the case AFAIK).

**Review by sudiptoguha — APPROVED** [review]
Looks good - perhaps there are followups in the name change?

File: src/main/java/org/opensearch/ad/AnomalyDetectorPlugin.java [hunk 2]
```diff
@@ -625,23 +626,23 @@ public PooledObject<LinkedBuffer> wrap(LinkedBuffer obj) {
             )
             .put(
                 StatNames.ANOMALY_DETECTORS_INDEX_STATUS.getName(),
-                new ADStat<>(true, new IndexStatusSupplier(indexUtils, AnomalyDetector.ANOMALY_DETECTORS_INDEX))
+                new ADStat<>(true, new IndexStatusSupplier(indexUtils, CommonName.CONFIG_INDEX))
```
[the remark this answers is not in the document]
Comment (kaituo) [reply src/main/java/org/opensearch/ad/AnomalyDetectorPlugin.java:629]: yes, will do that in a different PR.

File: src/main/java/org/opensearch/ad/constant/ADCommonName.java (hunk 1 above)
Comment (kaituo) [reply src/main/java/org/opensearch/ad/constant/ADCommonName.java:24]: To keep it backward-compatible, we will use opendistro as part of index name. For new indices (e.g., forecasting indices), we are gonna use opensearch.

**Review by jackiehanyang — APPROVED** [review]

**Review by sudiptoguha — COMMENTED** [review]
Same c comment as 901, looks good.
